### PR TITLE
Boiler bugs: Gas death causes, Other hive IFF gas

### PR DESCRIPTION
**Pull request by TeDGamer**
# About the pull request

Fixes a few things for boiler.

1. Non-normal hive boilers no longer generate normal hive gas. Meaning you can no longer FF your own hive (this also fixes death cause data for gases)
2. Fixes cause datas for neurotoxic gas and tail stab 
3. Another bug where unconscious people doesn't have the neurotoxin processed

smoke_system sets the cause_data on New. But set_up also sets up the cause data....if no parameter provided, it sets it to null. Was an issue since the gas based its hivenumber off the cause data mob

Had to add a useless named parameter to smoke chem gas to remove a warning, but it's unused for now but might be useful if we want death causes from chem gas

# Explain why it's good for the game

Fixing death causes is always good
Fixing boilers gassing their own hive to death because the gas wasn't the IFFed to the right hive is good for XvX
Bug fixing like usual where M.stat takes into account DEAD and UNCONSCIOUS when it should just be DEAD

# Changelog
:cl:
fix: Boilers from other hives generate proper colored IFFed gas
fix: Fixes some death causes with neurotoxin gas
fix: Neurotoxin stops processing only in dead mobs
/:cl:

### CODE REVIEW

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system/chemsmoke.dm [hunk 1]
```diff
@@ -39,7 +39,7 @@
 // Culls the selected turfs to a (roughly) circle shape, then calls smokeFlow() to make
 // sure the smoke can actually path to the turfs. This culls any turfs it can't reach.
 //------------------------------------------
-/datum/effect_system/smoke_spread/chem/set_up(datum/reagents/carry = null, n = 10, c = 0, loca, direct)
+/datum/effect_system/smoke_spread/chem/set_up(datum/reagents/carry = null, n = 10, c = 0, loca, direct, datum/cause_data/new_cause_data)
```
[the remark this answers is not in the document]
Comment (TeDGamer) [reply code/game/objects/effects/effect_system/chemsmoke.dm:42]: Edited to be similar to the parent smoke_spread setup.

image
Use the new cause data, else use the already stored cause data in effect_system datum

**Review by Drulikar — CHANGES_REQUESTED** [review]
I forced a merge since the branch was very old and was causing issues for me testing it - be sure to pull first.

The IFF gas change seems to be fine.

If the neuro is more potent from now affecting unconscious ends up to be too powerful, we can just tweak that line later - but its probably fine.

The cause data is better, but still not complete:
image

- Please fix cause data for acid gas ("Boiler" is the reason rather than acid gas)
- Please add the attacker to the cause data (second death in above says neurotoxin but doesn't attribute who's gas; for comparison see the third death from slashes, or the first from acid gas)

**Comment by Drulikar** [comment]
If you run into issues pulling because of names files, you can either:
`git reset --hard` and then `git pull`
or
Checkout a different branch e.g. `git checkout master --force`, delete your LOCAL version of the branch (either `git branch -d fix/boiler` or `git branch -D fix/boiler` if it warned you about losing changes to the files you don't care about), and then checkout the REMOTE version of the branch (e.g. `git fetch origin` then `git checkout -b fix/boiler origin/fix/boiler`) assuming your remote is `origin` and your branch actually is `fix/boiler` (that slash could be problematic).

**Comment by Watermelon914** [comment]
This PR has been inactive for some time, feel free to re-open (as a new PR so that it appears at the top of the tracker) when you decide to work on this again.

**Pull request closed without merging**